models/kustomization.go

```diff
@@ -25,7 +25,7 @@ type Kustomization struct {
 }
 
 // NewKustomization creates a new Kustomization object.
-func NewKustomization(resources, components []string) Kustomization {
+func NewKustomization(resources, components []string, namespace *string) Kustomization {
```

Comment:
Why not just make this a `string` instead of a `*string`?

Reply:
because I'd rather pass `nil` than an empty string, since this param is not always used. I thought it's more efficient that way.

Reply:
I'm not sure that efficiency is an important metric here. My concern is that now whenever we call this, even when passing a literal string, we need to decorate the parameter (`&var` or `&"a literal string"`). That second example, taking the address of a literal string, seems really non-obvious.

We're passing strings, rather than pointers-to-string, everywhere else (for role names, group names, component names, and  even namespaces elsewhere).

Reply:
I don't mind changing it. Let me quickly change it to pass by value then. :slightly_smiling_face:

Reply:
Done @larsks